course/grades.py

```diff
@@ -56,13 +57,15 @@
 from course.flow import adjust_flow_session_page_data
 from course.views import get_now_or_fake_time
 from course.constants import (
-        participation_permission as pperm,
+        participation_permission as pperm, MAX_EXTRA_CREDIT_FACTOR
         )
 
+csv_settings = RelateCSVSettingsInitializer()
```

Comment:
Can we do without this global variable somehow?

Reply:
(I'd much prefer a function, for example.)

Reply:
In fact, the `csv_setting` object created here is intended to be an app global objects, since I am creating the object with its properties to be [`cached_properties`](https://docs.djangoproject.com/en/2.0/ref/utils/#django.utils.functional.cached_property), each of which need to be calculated only once.

However, I agree it's not elegant to use it here in this way. What about initialize the `csv_settings` object, i.e.,

```python
csv_settings = RelateCSVSettingsInitializer()
```
in `utils.py`, and then import it in `grades.py`. Will that be better?